diff --git a/node_modules/react-native-worklets/android/CMakeLists.txt b/node_modules/react-native-worklets/android/CMakeLists.txt
index 1d949d6..83b7b04 100644
--- a/node_modules/react-native-worklets/android/CMakeLists.txt
+++ b/node_modules/react-native-worklets/android/CMakeLists.txt
@@ -94,7 +94,8 @@ endif()
 # build shared lib
 set_target_properties(worklets PROPERTIES LINKER_LANGUAGE CXX)
 
-target_link_libraries(worklets log ReactAndroid::jsi fbjni::fbjni)
+find_library(WORKLETS_CPP_SHARED c++_shared)
+target_link_libraries(worklets log ReactAndroid::jsi fbjni::fbjni ${WORKLETS_CPP_SHARED})
 
 if(ReactAndroid_VERSION_MINOR GREATER_EQUAL 76)
   target_link_libraries(worklets ReactAndroid::reactnative)
